src/AppInstallerCLICore/Workflows/UpdateFlow.h

```diff
@@ -39,13 +39,13 @@ namespace AppInstaller::CLI::Workflow
     // Outputs: None
     struct SelectSinglePackageVersionForInstallOrUpgrade : public WorkflowTask
     {
-        SelectSinglePackageVersionForInstallOrUpgrade(bool isUpgrade) :
-            WorkflowTask("SelectSinglePackageVersionForInstallOrUpgrade"), m_isUpgrade(isUpgrade) {}
+        SelectSinglePackageVersionForInstallOrUpgrade(SearchPurpose purpose) :
+            WorkflowTask("SelectSinglePackageVersionForInstallOrUpgrade"), m_searchPurpose(purpose) {}

```

Comment:
We should check the value passed in is expected.

THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), m_searchPurpose != Install && m_searchPurpose != Upgrade);